Makefile

```diff
@@ -0,0 +1,84 @@
+# Makefile for the charm store client.
+
+ifndef GOPATH
+$(warning You need to set up a GOPATH.)
+endif
+
+PROJECT := github.com/juju/charmstore-client
+PROJECT_DIR := $(shell go list -e -f '{{.Dir}}' $(PROJECT))
```

Comment:
This is a little opaque to me; I'm assuming it produces the path if it exists, and if not something that fails the match against CUR_DIR below?